src/board.pl

```diff
@@ -0,0 +1,60 @@
+:- dynamic board/4.
+:- dynamic board/3.
+:- dynamic board/2.
+
+% Initial pawn
+board(0, 1, pawn, white). %% a2
+board(0, 6, pawn, black). %% a7
+board(1, 1, pawn, white). %% b2
+board(1, 6, pawn, black). %% b7
+board(2, 1, pawn, white). %% c2
+board(2, 6, pawn, black). %% c7
+board(3, 1, pawn, white). %% d2
+board(3, 6, pawn, black). %% d7
+board(4, 1, pawn, white). %% e2
+board(4, 6, pawn, black). %% e7
+board(5, 1, pawn, white). %% f2
+board(5, 6, pawn, black). %% f7
+board(6, 1, pawn, white). %% g2
+board(6, 6, pawn, black). %% g7
+board(7, 1, pawn, white). %% h2
+board(7, 6, pawn, black). %% h7
+
+% Initial Bishop
+board(2, 0, bishop, white). %% c1
+board(2, 7, bishop, black). %% f1
+board(5, 0, bishop, white). %% c8
+board(5, 7, bishop, black). %% f8
+
+% Initial knight
+board(1, 0, knight, white). %% b1
+board(6, 0, knight, black). %% g1
+board(1, 7, knight, white). %% b8
+board(6, 7, knight, black). %% g8
+
+% Initial Rook
+board(0, 0, rook, white). %% a1
+board(7, 0, rook, black). %% a8
+board(0, 7, rook, white). %% h1
+board(7, 7, rook, black). %% h8
+
+% Initial Queen
+board(3, 0, queen, white). %% d1
+board(3, 7, queen, black). %% d8
+
+% Initial King
+board(4, 0, king, white). %% e1
+board(4, 7, king, black). %% e8
+
+updateBoard([Px, Py, Cx, Cy], Piece, Turn):-
+    removePiece(board(Px, Py, _, _)), 
+    removePiece(board(Cx, Cy, _, _)),
+    assert(board(Cx, Cy, Piece, Turn)), !.
+
+removePiece(Board):-
+    retract(Board).
+removePiece(_Board).
```

Comment:
```suggestion
removePiece(_).
```
Acho que aqui veio um typo

Reply:
Tipo assim, pode ser feito das duas maneiras, não é necessariamente um typo, no prolog a gnt pode utilizar tanto o _ quanto o _(Variável), a diferença é que no segundo caso estou especificando/documentando o que quero ignorar, fonte:
https://stackoverflow.com/questions/15416220/singleton-variables-in-prolog#:~:text=Singleton%20variables%20are%20useless%20in,will%20assume%20will%20be%20lost.

Reply:
![image](https://user-images.githubusercontent.com/74625814/181563767-9c691adf-5b79-42a4-8150-0aa84c740ed3.png)
Para melhor visualização.
https://www.swi-prolog.org/FAQ/SingletonVar.html

Reply:
Não sabia, massa.